### PR TITLE
feat: cli step 4 a02c iowaga thredds prior

### PR DESCRIPTION
**Pull request by cpaniaguam**
At the time of the push a service used was unavailable:

> syntax error, unexpected WORD_WORD, expecting SCAN_ATTR or SCAN_DATASET or SCAN_ERROR
> 
> context:
> <!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
> <html>
>     <head>
>         <title>503 Service Unavailable</title>
>     </head>
>     <body>
>         <h1>Service Unavailable</h1>
>         <p>
>             The server is temporarily unable to service your request due to maintenance downtime or capacity problems. Please try again later.
>         </p>
>     </body>
> </html>
> 
> Traceback (most recent call last):
> 
>   File "/home/cpaniaguam/icesat2-tracks/.venv/lib/python3.9/site-packages/xarray/backends/file_manager.py", line 211, in _acquire_with_cache_info
>     file = self._cache[self._key]
> 
>   File "/home/cpaniaguam/icesat2-tracks/.venv/lib/python3.9/site-packages/xarray/backends/lru_cache.py", line 56, in __getitem__
>     value = self._cache[key]
> 
> KeyError: [<class 'netCDF4._netCDF4.Dataset'>, ('https://tds3.ifremer.fr/thredds/dodsC/IOWAGA-WW3-FORECAST_GLOBMULTI_GLOB-30M_FIELD_NC_MARC_WW3-GLOB-30M',), 'r', (('clobber', True), ('diskless', False), ('format', 'NETCDF4'), ('persist', False)), '340f4832-a177-48f4-bab5-7e566c52c72b']

### CODE REVIEW

**Review by kmilo9999 — COMMENTED** [review]
This one runs in the CI. Should I go ahead with the review or wait for the other steps to merge?

**Comment by cpaniaguam** [comment]
@kmilo9999 Feel free to review it!

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 1]
```diff
@@ -17,526 +21,574 @@
...
-G1 = pd.DataFrame.from_dict(G1).T
-
-
-if hemis == "SH":
-    # DEFINE SEARCH REGION AND SIZE OF BOXES FOR AVERAGES
-    dlon_deg = 1  # degree range aroud 1st point
-    dlat_deg = 30, 5  # degree range aroud 1st point
-    dlat_deg_prior = 2, 1  # degree range aroud 1st point
+color_schemes.colormaps2(21)
+matplotlib.use("Agg")  # prevent plot windows from opening
```
Comment (kmilo9999) [inline src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:36]: it seems we can move this line right next to the 
`import matplotlib` line

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 2]
```diff
@@ -1,13 +1,17 @@
-import sys
+#!/usr/bin/env python3
+
 import datetime
+from pathlib import Path
 
 import h5py
 import pandas as pd
 import numpy as np
+import matplotlib
```
Comment (kmilo9999) [inline src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:9]: `import matplotlib; matplotlib.use("Agg")`

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 3]
```diff
@@ -17,526 +21,574 @@
...
-        "pdir1": "pdp1",
-        "pdir2": "pdp2",
-        "pdir3": "pdp3",
-        "pdir4": "pdp4",
-        "pdir5": "pdp5",
-    }
-)
+    all_beams = mconfig["beams"]["all_beams"]
 
+    file_name_base = "LOPS_WW3-GLOB-30M_"
```
Comment (kmilo9999) [inline src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:95]: this variables is use until line 331. We could probably move it to a better spot were makes sense to declare it

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 4]
```diff
@@ -17,526 +21,574 @@
...
+    # ECMWF forecast
+    data_url = "https://tds3.ifremer.fr/thredds/IOWAGA-WW3-FORECAST/IOWAGA-WW3-FORECAST_GLOBMULTI_GLOB-30M.xml"
+
+    cat = TDSCatalog(data_url)
+
+    ncss = cat.datasets[
+        "IOWAGA-WW3-FORECAST_GLOBMULTI_GLOB-30M_FIELD_NC_MARC_WW3-GLOB-30M"
+    ].remote_access(use_xarray=True)
+
+    var_list = [
```
Comment (kmilo9999) [inline src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:159]: This is a constant list. I think we can move it to the so called constant file we talked about

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 5]
```diff
@@ -17,526 +21,574 @@
...
-    F = M.figure_axis_xy(4, 3.5, view_scale=0.9, container=True)
-    plt.suptitle(
-        track_name_short + " | " + file_name_base[0:-1].replace("_", " "), y=1.3
+    # chunk data
+    IOWAGA = ncss[var_list]
+    IOWAGA["time"] = np.array([np.datetime64(k0) for k0 in IOWAGA.time.data]).astype(
+        "M8[h]"
+    )
+    IOWAGA = IOWAGA.rename(
+        name_dict={
```
Comment (kmilo9999) [inline src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:201]: same here. A list of constant values

**Review by kmilo9999 — CHANGES_REQUESTED** [review]
I added some comments. So many things we can do to improve the readability of the script

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py (hunk 1 above)
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:36]: Most linters complain about having multiple statements in one line -- l guess it makes things less readable. Also, we'd be mixing import statements with non-import statements.

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py (hunk 3 above)
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:95]: Moving it closer to L331.

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py (hunk 5 above)
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:201]: #65

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py (hunk 1 above)
Comment (hollandjg) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:36]: I agree, this should come after all the imports.

File: .gitignore [hunk 6]
```diff
@@ -45,6 +45,7 @@ logs/
 # Environments
 .env
 .venv
+.venv39/
```
Comment (hollandjg) [inline .gitignore:48]: This looks like a duplicate of line 51 in this file

File: src/icesat2_tracks/analysis_db/B02_make_spectra_gFT.py
Comment (hollandjg) [inline src/icesat2_tracks/analysis_db/B02_make_spectra_gFT.py:1]: Just noticed this: we should probably update the shebang lines to `#!/usr/bin/env python3` across the whole project.

**Review by hollandjg — CHANGES_REQUESTED** [review]
Hey Carlos, it feels like we're making a lot of progress here. A few comments and suggestions.

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 7]
```diff
@@ -17,527 +21,575 @@
...
+        # CFSR hindcast
+        # data_url = 'https://tds3.ifremer.fr/thredds/IOWAGA-WW3-HINDCAST/IOWAGA-GLOBAL_CFSR-WW3-HINDCAST_FULL_TIME_SERIE.xml'
+
+        # ECMWF forecast
+        data_url = "https://tds3.ifremer.fr/thredds/IOWAGA-WW3-FORECAST/IOWAGA-WW3-FORECAST_GLOBMULTI_GLOB-30M.xml"
+
+        cat = TDSCatalog(data_url)
+
+        ncss = cat.datasets[
+            "IOWAGA-WW3-FORECAST_GLOBMULTI_GLOB-30M_FIELD_NC_MARC_WW3-GLOB-30M"
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:155]: How to do this matching? The key probably depends on the data set pulled. I'll create an issue to maybe address this later. #114

File: src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py [hunk 8]
```diff
@@ -17,527 +21,575 @@
...
+
+        ## load WW3 data
+        # ECMWF hindcast
+        # data_url = 'https://tds3.ifremer.fr/thredds/IOWAGA-WW3-HINDCAST/IOWAGA-GLOBAL_ECMWF-WW3-HINDCAST_FULL_TIME_SERIE.xml'
+
+        # CFSR hindcast
+        # data_url = 'https://tds3.ifremer.fr/thredds/IOWAGA-WW3-HINDCAST/IOWAGA-GLOBAL_CFSR-WW3-HINDCAST_FULL_TIME_SERIE.xml'
+
+        # ECMWF forecast
+        data_url = "https://tds3.ifremer.fr/thredds/IOWAGA-WW3-FORECAST/IOWAGA-WW3-FORECAST_GLOBMULTI_GLOB-30M.xml"
```
[the remark this answers is not in the document]
Comment (cpaniaguam) [reply src/icesat2_tracks/analysis_db/A02c_IOWAGA_thredds_prior.py:150]: #114

**Review by kmilo9999 — COMMENTED** [review]
Looks good to me.  We should be able to merge if John doesnt have ant additional feedback.

**Review by kmilo9999 — APPROVED** [review]

**Review by hollandjg — APPROVED** [review]
Looks good to me!

**Pull request merged**